### PR TITLE
fixes #282

### LINKED ISSUE #282: Bug in Virustotal Oneshot
[2018-08-16 08:19:26,598: WARNING/ForkPoolWorker-2] Traceback (most recent call last):
[2018-08-16 08:19:26,598: WARNING/ForkPoolWorker-2] File "/home/sebdraven/sources/yeti/core/analytics_tasks.py", line 70, in single
[2018-08-16 08:19:26,598: WARNING/ForkPoolWorker-2] links = analytics.analyze(results.observable, results)
[2018-08-16 08:19:26,599: WARNING/ForkPoolWorker-2] File "/home/sebdraven/sources/yeti/plugins/analytics/public/virustotal_api.py", line 110, in analyze
[2018-08-16 08:19:26,599: WARNING/ForkPoolWorker-2] result['positives'] = json_result['positives']
[2018-08-16 08:19:26,599: WARNING/ForkPoolWorker-2] KeyError: u'positives'
[2018-08-16 08:19:26,617: INFO/ForkPoolWorker-2] Task core.analytics_tasks

### PR DESCRIPTION
**Pull request by sebdraven**
Fixe the bug if the observable is unknown by VT

### CODE REVIEW

File: plugins/analytics/public/virustotal_api.py [hunk 1]
```diff
@@ -80,57 +82,66 @@ def analyze(observable, results):
...
+                for k, v in create_hashes:
+                    new_hash = Hash.get_or_create(value=v)
+                    new_hash.tag(observable.get_tags())
+                    links.update(
+                        new_hash.active_link_to(observable, k,
+                                                'virustotal_query'))
+                result['scan_date'] = json_result['scan_date']
+            else:
+                result['scan_date'] = datetime.utcnow().strftime(
+                    '%Y-%m-%d %H:%M:%S')
```
Comment (tomchop) [inline plugins/analytics/public/virustotal_api.py]: I assume that `'%Y-%m-%d %H:%M:%S'` is the same format as `json_result['scan_date']`?

File: plugins/analytics/public/virustotal_api.py [hunk 2]
```diff
@@ -80,57 +82,66 @@ def analyze(observable, results):
...
-                o_isp = Company.get_or_create(name=json_result['as_owner'])
-                links.update(
-                    observable.active_link_to(
-                        o_isp, 'hosting', 'virustotal_query'))
-
-            if json_result.get('detected_urls'):
-                result['detected_urls'] = json_result['detected_urls']
-                for detected_url in json_result['detected_urls']:
-                    o_url = Url.get_or_create(value=detected_url['url'])
+            if json_result['response_code'] == 1:
```
Comment (tomchop) [inline plugins/analytics/public/virustotal_api.py]: You can also do the reverse logic to save an indentation: 

```
if json_result['response_code'] !- 1:
  return # Or do whatever it is you do when you get an invalid response
```

Additionally, this guard can also go before all the `isinstance` checks.

File: plugins/analytics/public/virustotal_api.py (hunk 1 above)
Comment (sebdraven) [reply plugins/analytics/public/virustotal_api.py]: Yep the same :)

File: plugins/analytics/public/virustotal_api.py [hunk 3]
```diff
@@ -128,9 +144,12 @@ def analyze(observable, results):
                 new_hash = Hash.get_or_create(value=v)
                 new_hash.tag(observable.get_tags())
                 links.update(
-                    new_hash.active_link_to(observable, k, 'virustotal_query'))
+                    new_hash.active_link_to(observable, k,
+                                            'virustotal_query'))
+
+            result['scan_date'] = json_result['scan_date']
 
         result['source'] = 'virustotal_query'
```
Comment (tomchop) [inline plugins/analytics/public/virustotal_api.py]: You should probably add this `source` key in the case `json_result['response_code'] != 1`

File: plugins/analytics/public/virustotal_api.py [hunk 4]
```diff
@@ -79,7 +80,18 @@ def analyze(observable, results):
...
+            result['permalink'] = None
+
+            observable.add_context(result)
+            return
+
         if isinstance(observable, Ip):
+
             # Parse results for ip
             if json_result.get('as_owner'):
                 result['Owner'] = json_result['as_owner']
```
Comment (tomchop) [inline plugins/analytics/public/virustotal_api.py]: please use lowercase keys in the dict (for consistency)

File: plugins/analytics/public/virustotal_api.py [hunk 5]
```diff
@@ -78,7 +78,12 @@ def analyze(observable, results):
         json_string = json.dumps(
             json_result, sort_keys=True, indent=4, separators=(',', ': '))
         results.update(raw=json_string)
-        result = {'raw': json_string}
+
+        result = dict()
```
Comment (tomchop) [inline plugins/analytics/public/virustotal_api.py]: Why all the whitespace? 

You can do
```
result = {
  'raw': json_string,
  'source: 'virustotal_query'
}

**Pull request merged**